packages/core/src/components/viewer-grid/viewer-grid.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  h,
+  Component,
+  Prop,
+  State,
+  Host,
+  Element,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import * as focusTrap from "focus-trap";
+
+type PanelSize = "small" | "medium" | "large";
+
+@Component({
+  tag: "dso-viewer-grid",
+  styleUrl: "viewer-grid.scss",
+  shadow: true,
+})
+export class ViewerGrid {
+  @Prop()
+  panelOpen = false;
+
+  @State()
+  panelSize: PanelSize = "large";
+
+  @Event()
+  closeAside!: EventEmitter<MouseEvent | KeyboardEvent>;
+
+  @Element()
+  host!: HTMLElement;
+
+  overlay!: HTMLDivElement;
+
+  overlaySlot!: HTMLDivElement;
+
+  trap!: focusTrap.FocusTrap;
+
+  shrinkPanel = () => {
+    this.panelSize = this.panelSize == "large" ? "medium" : "small";
+  };
+
+  expandPanel = () => {
+    this.panelSize = this.panelSize == "small" ? "medium" : "large";
+  };
+
+  updateFocusTrap() {
+    if (!this.trap) {
+      return;
+    }
+
+    if (this.panelOpen) {
+      this.trap.activate();
+      this.host.addEventListener("keydown", this.keyDownListener);
+    } else {
+      this.trap.deactivate();
+      this.host.removeEventListener("keydown", this.keyDownListener);
+    }
+  }
+
+  keyDownListener = (event: KeyboardEvent) => {
+    if (event.key != "Escape") {
+      return;
+    }
+
+    this.closeAside.emit(event);
+  };
+
+  connectedCallback() {
+    this.overlaySlot = this.host.querySelector(
+      "div[slot = 'overlay']"
+    ) as HTMLDivElement;
+  }
+
+  componentDidLoad() {
+    this.trap = focusTrap.createFocusTrap([this.overlay, this.overlaySlot], {
+      escapeDeactivates: false,
+      allowOutsideClick: true,
+    });
+
+    this.updateFocusTrap();
+  }
+
+  componentDidUpdate() {
+    this.updateFocusTrap();
+  }
+
+  render() {
+    return (
+      <Host class={this.panelSize}>
+        <div class={`dso-viewer-grid dso-viewer-grid-${this.panelSize}`}>
```

Comment:
Deze class (en element) was de host van het HTML/CSS component. Deze is nu komen te vervallen met het custom element `dso-viewer-grid`: Dit element kan volgens mij dus weg.

Reply:
Als ik die div weg haal dan werkt het in en uitklappen niet meer, toont de map en de overlay niet. Misschien dat @daverhelst daar beter nog even naar kan kijken.

Reply:
Klopt, het zal ook niet zomaar werken: Maar dit element is in ieder geval niet meer nodig. @daverhelst refactor jij het weg?

Reply:
@dion zie ook maar opmerking in https://github.com/dso-toolkit/dso-toolkit/pull/1436#pullrequestreview-892318043